src/test/java/com/github/cameltooling/lsp/internal/completion/KafkaTopicCompletionTest.java

```diff
@@ -139,9 +141,9 @@ private void initKafkaWithTopics(String topics) {
 		System.setProperty(KafkaTopicCompletionProvider.CAMEL_LANGUAGE_SERVER_KAFKA_CONNECTION_URL, connectionURL);
 	}
 
-	private List<CompletionItem> retrieveCompletionForKafkaTopicPosition() throws URISyntaxException, InterruptedException, ExecutionException {
+	private List<CompletionItem> retrieveCompletionForKafkaTopicPosition() throws URISyntaxException, InterruptedException, ExecutionException, TimeoutException {
 		CamelLanguageServer languageServer = initLanguageServer();
-		return getCompletionFor(languageServer, new Position(0, 17)).get().getLeft();
+		return getCompletionFor(languageServer, new Position(0, 17)).get(1, TimeUnit.SECONDS).getLeft();
```

Comment:
I think this part only puts a Timeout. Are you saying that the language server will try to avoid the timeout ensuring that it'll get the whatever is there in less than a second?

Reply:
The idea is that we do not want to have the completion call taking more than a second. From a user point of view when taking more than one second it is breaking the flow.  When creating a connection to a kafka cluster, in some cases, the connection hangs. I would like to ensure that when the connection hangs, the server is https://github.com/KaotoIO/vscode-kaoto/actions/runs/3436051234/jobs/5729130030#step:9:38 responding in less than a second and returning the "static" completion and not await for a Kafka cluster to answer.

Reply:
I see